[quant][graphmode][fx] Support quantizing FloatFunctional

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46634 [quant][graphmode][fx] Support quantizing FloatFunctional**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24438227](https://our.internmc.facebook.com/intern/diff/D24438227)

jerryzh168 committed:
[quant][graphmode][fx] Support quantizing FloatFunctional

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on test/quantization/test_quantize_fx.py:
```diff
@@ -1486,6 +1487,39 @@ def forward(self, x):
+            ns.call_method('dequantize')
+        ]
+        m = convert_fx(m)
+        self.checkGraphModuleNodes(m, expected_node_list=node_list)
```
makes sense, should we also add a numerical equivalency test for these for eager vs FX graph?

jerryzh168 replied:
sure, I can add a numerical test

vkuzo reviewed (approved):
looks great, thank you!

vkuzo commented on torch/quantization/quantize_fx.py:
```diff
@@ -12,6 +12,16 @@ def _check_is_graph_module(model):
             'Got type:' + str(type(model)) + ' Please make ' +
             'sure to follow the tutorials.')
 
+def _swap_ff_with_fxff(model):
```
would it make sense to also delete the old modules here?

jerryzh168 replied:
they are removed from model._modules when we reassign, is that necessary?

vkuzo replied:
optional IMO, I don't feel strongly.  It would be defensive if something outside of the model depends on the object.

jerryzh168 replied:
sure, done.

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support quantizing FloatFunctional"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24438227](https://our.internmc.facebook.com/intern/diff/D24438227)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support quantizing FloatFunctional"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24438227](https://our.internmc.facebook.com/intern/diff/D24438227)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support quantizing FloatFunctional"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24438227](https://our.internmc.facebook.com/intern/diff/D24438227)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support quantizing FloatFunctional"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24438227](https://our.internmc.facebook.com/intern/diff/D24438227)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support quantizing FloatFunctional"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24438227](https://our.internmc.facebook.com/intern/diff/D24438227)

[ghstack-poisoned]
